crates/sui-graphql-rpc/schema/current_progress_schema.graphql

```diff
@@ -1952,6 +1974,49 @@ type SharedInput {
 	mutable: Boolean!
 }
 
+"""
+The transaction accepted a shared object as input, but it was deleted before the transaction
+executed.
+"""
+type SharedObjectDelete {
+	"""
+	Address of the shared object.
```

Comment:
I am trying to understand the difference between Address of an object and the ID of an object (e.g., ID of the object being read.) This terminology sort of confuses me as I always think of them in a similar way. 
An address holds objects. But the `address` field of an object is actually the `ID`. 
So I assume in this case the address represents the actual address that holds this shared object?

Reply:
The `address` in this case is the ID of the shared object, only owned objects can be "held" by an address (that refers to a public key) in the way I think you mean -- I will clarify the comment.

Some backgroud:

`address` in Move is used to represent both the public keys that sign transactions (which can own objects), and the IDs of objects on-chain.  In the latter case, they are wrapped in the `ID` and/or `UID` types (again, in Move) to distinguish them programmatically.

In the RPC layer we made a choice not to create a separate scalar type for ObjectIDs -- we use `SuiAddress` for everything -- because in quite a few places, we needed to be able to provide an address that could either refer to an ObjectID or a public key, and that breaks the illusion that they are separate things, because you need to pick a shared type for that.  We were also influenced in this choice by the fact that:

- Adding extra scalars introduces friction for SDK developers.
- It's never possible for the same address to refer to both an Object and an Address, so using the same type to refer to both doesn't introduce the possibility of conflicts.